src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationContext.cs

```diff
@@ -98,6 +99,36 @@ public ValidationContext(object instance, IServiceProvider? serviceProvider, IDi
             ObjectInstance = instance;
         }
 
+        /// <summary>
+        ///     Construct a <see cref="ValidationContext" /> for a given object instance with
+        ///     a <paramref name="displayName" />, an optional <paramref name="serviceProvider" />,
+        ///     and an optional property bag of <paramref name="items" />.
+        /// </summary>
+        /// <param name="instance">The object instance being validated.  It cannot be null.</param>
+        /// <param name="displayName">The display name associated with the object instance.</param>
+        /// <param name="serviceProvider">
+        ///     Optional <see cref="IServiceProvider" /> to use when <see cref="GetService" /> is called.
+        ///     If it is null, <see cref="GetService" /> will always return null.
+        /// </param>
+        /// <param name="items">
+        ///     Optional set of key/value pairs to make available to consumers via <see cref="Items" />.
+        ///     If null, an empty dictionary will be created.  If not null, the set of key/value pairs will be copied into a
+        ///     new dictionary, preventing consumers from modifying the original dictionary.
+        /// </param>
+        /// <exception cref="ArgumentNullException">When <paramref name="instance" /> is <c>null</c></exception>
+        /// <remarks>
+        ///     This constructor is trim-safe because it does not use reflection to resolve
+        ///     the Type of the <paramref name="instance" /> to support setting the DisplayName.
+        /// </remarks>
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = TrimSafeConstructorJustification)]
```

Comment:
It is unusual for us to be suppressing public APIs like this  unless we guard access to it via an annotation (e.g.we sometimes suppress instance methods if the constructor itself has been annotated). 

In this case though, I think this needs to be refactored somewhat so that it does not chain with a constructor that is annotated.